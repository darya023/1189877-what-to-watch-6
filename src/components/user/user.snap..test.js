import React from 'react';
import {render, screen} from '@testing-library/react';
import {Router, Switch, Route} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import userEvent from '@testing-library/user-event';
import * as redux from 'react-redux';
import User from './user.jsx';

const mockStore = configureStore({});

let history;

describe(`Test for User`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });

  it(`User should render correctly if unauthorized`, () => {
    const store = mockStore({
      USER: {
        authorizationStatus: false,
        user: null,
      },
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <User />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  it(`User should render correctly if unauthorized`, () => {
    const fakeUser = {
      id: `1`,
      name: `User`,
      image: ``,
      email: `test@test.ru`
    };
    const store = mockStore({
      USER: {
        authorizationStatus: true,
        user: fakeUser,
      },
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <User />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
