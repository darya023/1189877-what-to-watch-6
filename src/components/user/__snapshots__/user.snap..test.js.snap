// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for User User should render correctly if unauthorized 1`] = `
<div>
  <div
    class="user-block"
  >
    <a
      class="user-block__link"
      href="/login"
    >
      Sign in
    </a>
  </div>
</div>
`;

exports[`Test for User User should render correctly if unauthorized 2`] = `
<div>
  <div
    class="user-block"
  >
    <div
      class="user-block__avatar"
    >
      <a
        href="/mylist"
      >
        <img
          alt="User avatar"
          height="63"
          src=""
          width="63"
        />
      </a>
    </div>
    <div>
      test@test.ru
    </div>
  </div>
</div>
`;
